matrix.h

```diff
@@ -0,0 +1,114 @@
+#ifndef MATRIX_H
+#define MATRIX_H
+
+
+
+using namespace std;
+
+class Matrix {
+
+    private:
+        double* mat;
+        double rows;
+        double cols;       
+
+    public:
+
+        int size;
+
+        Matrix(double _rows, double _cols){
+            rows = _rows;
+            cols = _cols;
+
+            size = _rows * _cols;
+
+            mat = new double[size];
+
+            for (int i = 0; i < _rows * _cols; i++){
+               mat[i] = 0;
+            }
+        };
+        Matrix(double _rows, double _cols,double* matrix){
+            rows = _rows;
+            cols = _cols;
+
+            size = _rows * _cols;
+
+            mat = new double[size];
+
+            for(int i = 0; i < size; i++){
+                *(mat + i) = *(matrix + i);
+            };
+
+        };
+
+        // Scalar Math
+        // Matrix operator*(double rhs){
+        //     Matrix result(rows, cols);
+
+        //     for (double i=0; i<rows; i++) {
+        //         for (double j=0; j<cols; j++) {
+        //             result.mat[i][j] = mat[i][j] * rhs;
+        //         }
+        //     }
+
+        //     return result;
+        // };
+        // Matrix operator/(double rhs){
+        //     Matrix result(rows, cols);
```

Comment:
Your matrix class looks pretty clean. everything seems to be straight forward. good job.